RELEASES.md

```diff
@@ -1,3 +1,100 @@
+Version 1.82.0 (2024-10-17)
+==========================
+
+<a id="1.82.0-Language"></a>
+
+Language
+--------
+- [Don't make statement nonterminals match pattern nonterminals](https://github.com/rust-lang/rust/pull/120221/)
+- [Implement lint against ambiguous negative literals](https://github.com/rust-lang/rust/pull/121364/)
+- [Patterns matching empty types can now be omitted in common cases](https://github.com/rust-lang/rust/pull/122792)
+- [size_of_val_raw: for length 0 this is safe to call](https://github.com/rust-lang/rust/pull/126152/)
+- [Reorder trait bound modifiers *after* `for<...>` binder in trait bounds](https://github.com/rust-lang/rust/pull/127054/)
+- [Stabilize opaque type precise capturing (RFC 3617)](https://github.com/rust-lang/rust/pull/127672)
+- [Stabilize explicit opt-in to unsafe attributes](https://github.com/rust-lang/rust/pull/128771)
+
+
+<a id="1.82.0-Compiler"></a>
+
+Compiler
+--------
+- [Promote riscv64gc-unknown-linux-musl to tier 2](https://github.com/rust-lang/rust/pull/122049)
+- [Improved diagnostics to explain why a pattern is unreachable](https://github.com/rust-lang/rust/pull/128034)
+- [The compiler now triggers the unreachable code warning properly for async functions that don't return/are `-> !`](https://github.com/rust-lang/rust/pull/128443)
+- [Promote `wasm32-wasip2` to Tier 2.](https://github.com/rust-lang/rust/pull/126967/)
+
+
+<a id="1.82.0-Libraries"></a>
+
+Libraries
+---------
+- [Implement DoubleEnded and ExactSize for Take<Repeat> and Take<RepeatWith>](https://github.com/rust-lang/rust/pull/106943/)
+- [Stabilize Wasm relaxed SIMD](https://github.com/rust-lang/rust/pull/117468/)
+- [Stabilize const `{integer}::from_str_radix` i.e. `const_int_from_str`](https://github.com/rust-lang/rust/pull/124941/)
+- [CommandExt::before_exec: deprecate safety in edition 2024](https://github.com/rust-lang/rust/pull/125970/)
+- [CloneToUninit impls](https://github.com/rust-lang/rust/pull/126877/)
```

Comment:
```suggestion
- [`CloneToUninit` impls](https://github.com/rust-lang/rust/pull/126877/)
```

Reply:
This one is also kinda terse, but I don't understand the context enough to better reflect what changed here. Seems internal though, so I suppose that's fine.

Reply:
This allows calling `Rc<T>/Arc<T>::make_mut` for `T` = `Path`, `OsStr`, and `CStr` on stable.

In 1.81 we made a similar change for just `[T]` and `str` and the relnotes list that as

> Generalize `{Rc,Arc}::make_mut()` to unsized types.

so perhaps this one could be

```suggestion
- [Generalize `{Rc,Arc}::make_mut()` to `Path`, `OsStr`, and `CStr`.](https://github.com/rust-lang/rust/pull/126877/)
```